### PR TITLE
fix #1872 Avoid classloader leak due to backtrace field in static throwables

### LINKED ISSUE #1872: Classloader leak with Throwable in static fields
In dynamic classloader environments, the use of Throwable in static fields may cause a classloader leak in the backtrace field. This field is set in runtime at native code with an array that are instances of Class. This may cause a Classloader which should be disposed to be retained.

In other [projects](https://github.com/eclipse/jetty.project/issues/868), an error without a writable stacktrace was used for this cases. 

I've opened a PR for this:
https://github.com/reactor/reactor-core/pull/1871

Let me know.

Thanks!

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@fsgonz Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/reactor?repositoryId=reactor/reactor-core&pullRequestId=1871)!

[Click here](https://cla.pivotal.io/sync/reactor?repositoryId=reactor/reactor-core&pullRequestId=1871) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@fsgonz Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/reactor?repositoryId=reactor/reactor-core&pullRequestId=1871)!

**Comment by bsideup** [comment]
Hi @fsgonz,

Could you please update the description and explain a bit why `Throwable` creates a leak here, given that it is JVM's class and not user's?

Thanks!

**Comment by fsgonz** [comment]
@bsideup I've opened an issue referencing this: [#1872 ](https://github.com/reactor/reactor-core/issues/1872)

File: reactor-core/src/main/java/reactor/core/TerminatedThrowable.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.core;
+
+/**
+ * Terminated Throwable to avoid classlaoder leaks in backtrace field
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/core/TerminatedThrowable.java]: typo `classlaoder`

File: reactor-core/src/main/java/reactor/core/TerminatedThrowable.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ */
+public class TerminatedThrowable extends Error {
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/core/TerminatedThrowable.java]: I would rename to `StaticThrowable` or `ConstantThrowable`

**Comment by simonbasle** [comment]
@fsgonz I think this would be beneficial in 3.2.x, can you rework it to be based on branch `3.2.x`? the best way to do that is by interactive rebase:

```bash
git checkout leak
git rebase -i 3.2.x
```
in the opened editor, be careful to remove all the commit lines except the last one (which is this PR's commit). save and close the editor.
```bash
git push --force-with-lease
```
then in this PR, click `Edit` and click on the base branch, a dropdown will let you re-target the PR against `3.2.x`.
🙇

**Comment by simonbasle** [comment]
@fsgonz I will polish all this myself, since targeting 3.2 means that this has to be ready by tomorrow

**Comment by simonbasle** [comment]
actually I don't have rights to modify your PR, I'll close and redo it. Thanks for reporting!

**Comment by fsgonz** [comment]
Sorry, I've just read your comments. Do you want me to perform any changes?

**Comment by simonbasle** [comment]
No problem, I was just concerned about the timing so I opened a new PR. I did cherry-pick your change thoug. See #1876

**Pull request closed without merging**